consensus/beacon/consensus.go

```diff
@@ -405,6 +413,7 @@ func (beacon *Beacon) FinalizeAndAssemble(chain consensus.ChainHeaderReader, hea
 			return nil, fmt.Errorf("nil parent header for block %d", header.Number)
 		}
 
+		state.Database().LoadTransitionState(parent.Root)
```

Comment:
When producing the proof, the pre-tree is re-opened. Unfortunately, the `OpenTrie` function needs the pointers, who have to be reloaded. This is a bit tacky and could be avoided if the pre tree was opened before, e.g. in `Prepare`.